docs/defining_new_slo.md

```diff
@@ -0,0 +1,191 @@
+# Defining new SLO
+
+> In case you don't know what SLO is we recommend you to read [the SRE workbook](https://sre.google/workbook/implementing-slos/).
+> Shortcuts used:
+>   - **Service Level Indicator(SLI)**: Measurable phenomenon that describes some crucial characteristic of the observed service(ideally in %).
+>   - **Service Level Objective(SLO)**: Desired level of quality for given SLI.
+>   - **Service Level Agreement(SLA)**: Agreement between service provider and user about what happens if the SLO is not matched.
+>   - **Error budget(EB)**: Acceptable amount of errors over a specified time window based on the SLO.
+
+Each service having any consumers should have defines SLO.
+For example if your team manages set of microservices,
+most probably you would set and measure SLO for any interaction outside your team components(other team components, end users).
+Keep in mind that service cannot have higher SLO than lowest of services it depends on.
+
+# 1. Defining SLI
+Do a brainstorming for both developers and product owners to
+think of the important functionality and characteristics of your service
+and what matters to your customers. These will be your SLIs. 
+You should be able to represent those as a % of success or failure for the further computation.
+> Hint: Do not try to overcomplicate things right from the beginning and stop it from actually computing at least some SLIs.
+>       You can iterate on the accuracy later on.
+
+## SLI computation approaches
+There are 2 main approaches to measure the SLI
+
+### Time based
+This SLI would be based on some time interval and percentage of the time the service meets the criteria.
+Most common use-case would be an uptime computed as percentage of the time the service is operational(it might be tricky to tell what functional means).
```

Comment:
Google presented a more complex implementation of time-based SLO with:
- computing per-user uptime
- computing resulting uptime for large number of SLO time-windows

See [the talk](https://www.usenix.org/conference/nsdi20/presentation/hauer) for details.